R/viz.R

```diff
@@ -2082,225 +2011,362 @@ splendidHeatmap = function(this_matrix,
-      my_colours = append(my_colours, list(c(this_metadata_column, "NA"="#BDBDC1FF")))
-      names(my_colours) = these_names
+    this_metadata_column = get_gambl_colours(metadataColumns[i])
+    if (sum(is.na(names(this_metadata_column[unique(these_samples_metadata[,metadataColumns[i]])]))) <= 1 &
+       length(unique(these_samples_metadata[,metadataColumns[i]])) > 1){
+        these_names = c(these_names, metadataColumns[i])
+        my_colours = append(my_colours, list(c(this_metadata_column, "NA" = "#BDBDC1FF")))
+        names(my_colours) = these_names
     }
   }
 
-  my_colours <- c(custom_colours, my_colours)
+  my_colours = c(custom_colours, my_colours)
 
   col_fun=circlize::colorRamp2(c(0, 0.5, 1), c("blue", "white", "red"))
   for(exp in numericMetadataColumns){
     my_colours[[exp]] = col_fun
   }
 
-  # get all features
-  w <- importance_values[,comparison_groups]
-  w <- as.data.frame(w) %>%
+  #get all features
+  w = importance_values[,comparison_groups]
+  w = as.data.frame(w) %>%
     dplyr::mutate_if(is.character,as.numeric)
 
-
-
-  # extract most important features, while taking the feature with highest weight for a particular cluster if it was seen before for other cluster with lower weight
-  FEATURES <- w[,1] %>%
+  #extract most important features, while taking the feature with highest weight for a particular cluster if it was seen before for other cluster with lower weight
+  FEATURES = w[,1] %>%
     as.data.frame() %>% 
-    `rownames<-`(rownames(w)) %>%
+    `rownames = `(rownames(w)) %>%
     dplyr::arrange(desc(.)) %>%
     head(., max_number_of_features_per_group) %>%
-    rownames_to_column(., var="Feature") %>%
-    dplyr::mutate(group=comparison_groups[1])
+    rownames_to_column(., var = "Feature") %>%
+    dplyr::mutate(group = comparison_groups[1])
   for (i in 2:length(comparison_groups)){
-    FEATURES <- rbind(as.data.frame(FEATURES),
-                    w[,i] %>% as.data.frame() %>%
-                      `rownames<-`(rownames(w)) %>%
-                      dplyr::arrange(desc(.)) %>%
-                      head(., max_number_of_features_per_group+3) %>%
-                      rownames_to_column(., var="Feature") %>%
-                      dplyr::mutate(group=comparison_groups[i])) %>%
-    dplyr::group_by(Feature) %>%
-    dplyr::filter(. == max(.)) %>%
-    dplyr::arrange(group)
-  }
-  FEATURES <- as.data.frame(FEATURES)
-
-  mat <- this_matrix %>%
+    FEATURES = rbind(as.data.frame(FEATURES), w[,i] %>% 
+      as.data.frame() %>%
+      `rownames = `(rownames(w)) %>%
+       dplyr::arrange(desc(.)) %>%
+       head(., max_number_of_features_per_group + 3) %>%
+       rownames_to_column(., var = "Feature") %>%
+       dplyr::mutate(group = comparison_groups[i])) %>%
+       dplyr::group_by(Feature) %>%
+       dplyr::filter(. == max(.)) %>%
+       dplyr::arrange(group)
+  }
+  FEATURES = as.data.frame(FEATURES)
+
+  mat = this_matrix %>%
     merge(., metadata_df %>%
```

Comment:
can you please change here `merge` to be `base::merge`? This function was developed in R3.6 and I found that in R4 never version of some package gets conflicted with this, causing all `-` characters in column names to become `.`. I can confirm I tested implementation with `base::merge` that solves this, and it is better to refer to function explicitly. Thanks!